src/bundle/Resources/public/js/scripts/fieldType/base/base-rich-text.js

```diff
@@ -263,38 +263,50 @@
             const counterWrapper = container.parentElement.querySelector('.ez-character-counter');
 
             if (counterWrapper) {
-                const countableTags = this.getCountableTags(editorHtml);
                 const wordWrapper = counterWrapper.querySelector('.ez-character-counter__word-count');
                 const charactersWrapper = counterWrapper.querySelector('.ez-character-counter__character-count');
-                let characterCount = 0;
-                let wordCount = 0;
+                let words = this.getTextNodeValues(editorHtml);
 
-                countableTags.forEach((tag) => {
-                    const sanitizedText = this.cleanWhiteCharacters(tag.innerText);
-
-                    wordCount += sanitizedText ? sanitizedText.split(' ').length : 0;
-                    characterCount += sanitizedText.length;
-                });
+                wordWrapper.innerText = words.length;
+                charactersWrapper.innerText = words.join(' ').length;
+            }
+        }
 
-                wordWrapper.innerText = wordCount;
-                charactersWrapper.innerText = characterCount;
+        getTextNodeValues(node) {
+            const TEXT_NODE = 3;
+            const notAllowedAttributes = ['ezconfig', 'ezvalue'];
+            let values = [];
+
+            const pushValue = (node) => {
+                if (node.nodeType === TEXT_NODE &&
+                    !notAllowedAttributes.includes(node.parentNode.getAttribute('data-ezelement'))
```

Comment:
Not sure if we should iterate over `notAllowedAttributes`. If the element is in not allowed we should skip it. So this check should be in `iterateThroughChildNodes` with the early return when has `ezconfig` attribute (less HTML to go through).